src/libYARP_OS/include/yarp/os/FixedSizeBuffersManager.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_OS_FIXEDSIZEBUFFERSMANAGER_H
+#define YARP_OS_FIXEDSIZEBUFFERSMANAGER_H
+
+#include <yarp/os/Mutex.h>
+#include <vector>
+#include <stdexcept>
+
+
+// Defined in this file:
+namespace yarp { namespace os { template <typename T> class FixedSizeBuffersManager; }}
+namespace yarp { namespace os { template <typename T> class Buffer; }}
+
+
+namespace yarp {
+namespace os {
+
+
+/**
+* Buffer contains info about a buffer of type T and it is used to exchange information
+* with yarp::os::FixedSizeBuffersManager.
+*/
+
+template <typename T>
+class Buffer
+{
+friend class FixedSizeBuffersManager<T>;
```

Comment:
I personally dislike friendship among c++ classes, but it is debated if it is or it is not a bad practice, it depends on the specific usage.
Why do you need it here?
If it is to access to the members of `Buffer` class, I'd advice to add setters and getters.